src/swe/swe_petsc.c

```diff
@@ -10,30 +10,75 @@ static const PetscReal GRAVITY = 9.806;
 
 /// For computing fluxes, allocates structs to hold values left and right
 /// of internal and boundary edges. This must be called before CreatePetscSWESource.
-PetscErrorCode CreatePetscSWEFlux(PetscInt num_internal_edges, PetscInt num_boundaries, RDyBoundary boundaries[num_boundaries], void **petsc_rhs) {
+PetscErrorCode CreatePetscSWEFlux(RDyCells *cells, RDyEdges *edges, PetscInt ncomp, PetscInt num_internal_edges, PetscInt num_boundaries,
+                                  RDyBoundary boundaries[num_boundaries], PetscBool ceed_enabled, void **petsc_rhs) {
```

Comment:
Things are getting pretty busy in these function signatures, but I think this can be addressed in the upcoming refactor/cleanup.